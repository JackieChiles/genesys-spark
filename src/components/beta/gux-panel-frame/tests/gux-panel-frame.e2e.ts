import { newSparkE2EPage, a11yCheck } from '../../../../../tests/e2eTestUtils';

const axeExclusions = [];

describe('gux-panel-frame', () => {
  const headerSlot = '<div slot="header"><h1>Header</h1></div>';
  const bodySlot = '<div slot="body"><p>Boby</p></div>';
  const footerSlot = '<div slot="footer"><footer>footer</footer></div>';
  const invalidSlot =
    '<div slot="invalid" id="invalid"><span>Invalid</span></div>';

  describe('#render', () => {
    [
      `<gux-panel-frame-beta>${headerSlot}${bodySlot}${footerSlot}</gux-panel-frame-beta>`,
      `<gux-panel-frame-beta>${bodySlot}${footerSlot}</gux-panel-frame-beta>`,
      `<gux-panel-frame-beta>${headerSlot}${bodySlot}</gux-panel-frame-beta>`,
      `<gux-panel-frame-beta>${headerSlot}${footerSlot}</gux-panel-frame-beta>`,
      `<gux-panel-frame-beta>${headerSlot}</gux-panel-frame-beta>`,
      `<gux-panel-frame-beta>${headerSlot}${bodySlot}${invalidSlot}${footerSlot}</gux-panel-frame-beta>`
    ].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newSparkE2EPage({ html });
        const element = await page.find('gux-panel-frame-beta');
        await a11yCheck(page, axeExclusions);

        expect(element.innerHTML).toMatchSnapshot();
      });
    });
  });
});
