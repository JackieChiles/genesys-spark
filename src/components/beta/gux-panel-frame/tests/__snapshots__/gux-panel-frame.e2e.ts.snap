// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-panel-frame #render should render component as expected (1) 1`] = `"<!----><section class=\\"gux-panel-container\\"><header class=\\"gux-panel-header\\"><div slot=\\"header\\"><h1>Header</h1></div></header><div class=\\"gux-panel-body\\"><div slot=\\"body\\"><p>Boby</p></div></div><footer class=\\"gux-panel-footer\\"><div slot=\\"footer\\"><footer>footer</footer></div></footer></section>"`;

exports[`gux-panel-frame #render should render component as expected (2) 1`] = `"<!----><section class=\\"gux-panel-container\\"><div class=\\"gux-panel-body\\"><div slot=\\"body\\"><p>Boby</p></div></div><footer class=\\"gux-panel-footer\\"><div slot=\\"footer\\"><footer>footer</footer></div></footer></section>"`;

exports[`gux-panel-frame #render should render component as expected (3) 1`] = `"<!----><section class=\\"gux-panel-container\\"><header class=\\"gux-panel-header\\"><div slot=\\"header\\"><h1>Header</h1></div></header><div class=\\"gux-panel-body\\"><div slot=\\"body\\"><p>Boby</p></div></div></section>"`;

exports[`gux-panel-frame #render should render component as expected (4) 1`] = `"<!----><section class=\\"gux-panel-container\\"><header class=\\"gux-panel-header\\"><div slot=\\"header\\"><h1>Header</h1></div></header><footer class=\\"gux-panel-footer\\"><div slot=\\"footer\\"><footer>footer</footer></div></footer></section>"`;

exports[`gux-panel-frame #render should render component as expected (5) 1`] = `"<!----><section class=\\"gux-panel-container\\"><header class=\\"gux-panel-header\\"><div slot=\\"header\\"><h1>Header</h1></div></header></section>"`;

exports[`gux-panel-frame #render should render component as expected (6) 1`] = `"<!----><div slot=\\"invalid\\" id=\\"invalid\\" hidden=\\"\\"><span>Invalid</span></div><section class=\\"gux-panel-container\\"><header class=\\"gux-panel-header\\"><div slot=\\"header\\"><h1>Header</h1></div></header><div class=\\"gux-panel-body\\"><div slot=\\"body\\"><p>Boby</p></div></div><footer class=\\"gux-panel-footer\\"><div slot=\\"footer\\"><footer>footer</footer></div></footer></section>"`;
